sql/statement.go

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (c) 2008-2022, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package sql
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/hazelcast/hazelcast-go-client/internal/check"
+	ihzerrors "github.com/hazelcast/hazelcast-go-client/internal/hzerrors"
+)
+
+const (
+	// duplication of constants from internal driver package.
+	// todo make sure they are same with the originals with tests.
+	defaultCursorBufferSize int32 = 4096
+	defaultTimeoutMillis    int64 = -1
+	defaultSchema                 = ""
+)
+
+/*
+Statement represents an SQL Statement. Use NewStatement to benefit
+from the default settings.
+Fields are read once before the execution is started.
+Changes to fields do not affect the behavior of already running statements.
+*/
+type Statement struct {
```

Comment:
There seems to be a field alignment warning. Could you run `make check`?

Reply:
Sure fixed on https://github.com/hazelcast/hazelcast-go-client/pull/755/commits/39a84bbc27b6aea101f25b0c5dc50ba1bb1175b8